yt_dlp/extractor/disney.py

```diff
@@ -71,20 +76,39 @@ class DisneyIE(InfoExtractor):
     }, {
         'url': 'http://disneyjunior.disney.com/galactech-the-galactech-grab-galactech-an-admiral-rescue',
         'only_matching': True,
+    }, {
+        # Grill.burger
+        'url': 'http://kids.disney.co.jp/video/3178',
+        'info_dict': {
+            'id': '5e926e4097ecfebd6b4f2a72',
+            'ext': 'mp4',
+            'title': '本編第1話（ほんぺんだい1わ） マーベル アルティメット・スパイダーマン\u3000ウェブ・ウォーリアーズ',
+            'upload_date': '20221018',
+            'description': 'md5:7978279f5a9b79e350613b8e0a507c24',
+            'duration': 1320,
+            'thumbnail': 'https://lumiere-a.akamaihd.net/v1/images/26412_32232_31532_1_35441_12510_12540_12505_12523_12450_1e89505b.jpeg?region=0%2C0%2C1280%2C720',
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': True,
+        },
     }]
 
+    # https://www.starwars.com times out with the default user-agent
+    _USER_AGENT = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:132.0) Gecko/20100101 Firefox/132.0'
+
     def _real_extract(self, url):
         domain, video_id, display_id = self._match_valid_url(url).groups()
         if not video_id:
-            webpage = self._download_webpage(url, display_id)
+            webpage = self._download_webpage(url, display_id, headers={'user-agent': self._USER_AGENT})
             grill = re.sub(r'"\s*\+\s*"', '', self._search_regex(
                 r'Grill\.burger\s*=\s*({.+})\s*:',
                 webpage, 'grill data'))
-            page_data = next(s for s in self._parse_json(grill, display_id)['stack'] if s.get('type') == 'video')
-            video_data = page_data['data'][0]
+            video_data = next(traverse_obj(s, ('data', 0, 'video')) or traverse_obj(s, ('data', 0)) for s in self._parse_json(grill,
+                                                                                                                              display_id)['stack'] if s.get('type') in ('video', 'flexcontenthero'))
```

Comment:
you can use a single traversal call for this, and please fix the formatting

Reply:
Oh, nice 👍 Okay, combined the `traverse_obj` calls now. Regarding the formatting, if you're talking about the awkward line break, I think that happened when I ran ruff on it before the last commit. I moved it all back to the same line.